suite-native/device/src/hooks/useDeviceChecks.ts

```diff
@@ -0,0 +1,50 @@
+import { useSelector } from 'react-redux';
+
+import { selectIsFirmwareAuthenticityCheckDismissed } from '@suite-common/wallet-core';
+
+import {
+    selectHasFirmwareAuthenticityCheckHardFailed,
+    selectIsDeviceAuthenticityCheckFailed,
+    selectIsEntropyCheckEnabledAndFailed,
+} from '../selectors';
+
+export const useDeviceChecks = (isDeviceCompromisedModalFocused: boolean) => {
+    const hasFirmwareAuthenticityCheckHardFailed = useSelector(
+        selectHasFirmwareAuthenticityCheckHardFailed,
+    );
+    const isFirmwareAuthenticityCheckDismissed = useSelector(
+        selectIsFirmwareAuthenticityCheckDismissed,
+    );
+    const shouldDisplayDeviceAuthenticityError = useSelector(selectIsDeviceAuthenticityCheckFailed);
+    const shouldDisplayEntropyCheckError = useSelector(selectIsEntropyCheckEnabledAndFailed);
+
+    const shouldDisplayFirmwareAuthenticityError =
+        hasFirmwareAuthenticityCheckHardFailed && !isFirmwareAuthenticityCheckDismissed;
```

Comment:
I wouldn't use _should_ in variable names because of its semantics. What about something like `isDeviceAuthenticityErrorDisplayable`?

Reply:
I was just moving this from other place, but I agree, let's get rid of the `should`